pdf/pdfcanvas.cpp

```diff
@@ -425,21 +433,45 @@ QSGNode* PDFCanvas::updatePaintNode(QSGNode *node, QQuickItem::UpdatePaintNodeDa
         bool loadPage = page.rect.intersects(loadedArea);
         bool showPage = page.rect.intersects(visibleArea);
 
-        if (loadPage) {
-            if ((!page.texture)
-                || (page.texture && page.texture->textureSize().width() != d->renderWidth)) {
-                if (page.requested && showPage) {
-                    priorityRequests << QPair<int, int>(i, d->renderWidth);
-                } else if (!page.requested) {
-                    d->document->requestPage(i, d->renderWidth, window());
-                    page.requested = true;
-                }
+        QRectF textureLimit = {
+            0, 0,
+            float(2. * window()->width()) / page.rect.width(),
+            float(2. * window()->height() / page.rect.height())
+        };
+        
+        QRectF showableArea = {
+            visibleArea.x() - float(window()->width() / 4.),
+            visibleArea.y() - float(window()->height() / 4.),
+            visibleArea.width() + float(window()->width() / 2.),
+            visibleArea.height() + float(window()->height() / 2.)
+        };
+        showableArea = page.rect.intersected(showableArea);
+        // remove one pixel margin to avoid rounding errors.
+        showableArea = QRectF((showableArea.x() + 1) / page.rect.width(),
+                              (showableArea.y() - page.rect.y() + 1) / page.rect.height(),
+                              (showableArea.width() - 2) / page.rect.width(),
+                              (showableArea.height() - 2) / page.rect.height());
+
```

Comment:
I think you're right. At the beginning , I wanted to keep the textureArea in reduced, because I was thinking for the annotation partial rendering, where we go patching a piece of the page, and we want this patch to be applied on different zoom levels. But that's wrong in fact. When zooming, the patches will be removed and the full page re-rendered.

So the textureArea can be kept in pixel, synchronized with the page width, less headache, I agree. I'll change this when I have a bit of free time.